### ISSUE TITLE
mdapt dedithering doesn't blend very well with bigger grids

### ISSUE THREAD

**Issue by whatwwkd4496** [libretro/common-shaders#156]
![Original](https://i.imgur.com/GNtpMbd.png)
![mdapt](https://i.imgur.com/Zoh53UM.png)
![gdapt](https://i.imgur.com/ulLNWQy.png)
![cbod](https://i.imgur.com/iS98Lvh.png)

**Comment by whatwwkd4496** [comment]
i've done some manual tinkering of this scene to get an idea how it would turned out
![manual tinkering](https://i.imgur.com/9QIjNz0.png)
manual tinkering by splitting odd and even pixels verticaly and horizontaly and further splitting halved pixels to quarter size to get 4 split images then blend them into two final look.

**Comment by hizzlekizzle** [comment]
yeah, mdapt is pretty cautious about dithering detection. gdapt is from the same author as mdapt and is designed to catch more patterns at the cost of more false-positives.

GTUv50 is another good shader for handling dithering. You can drop the horizontal/vertical signal bandwidth until the dithering starts to blend. It's basically a blur that you can fine-tune:
![image](https://user-images.githubusercontent.com/2805586/69876580-96e8a200-1285-11ea-9575-396df95be210.png)

**Comment by whatwwkd4496** [comment]
could you make an edge preserving dedithering with two variants, one is for graphics priority with more aggressive blending with more errors, and another is for text priority with subtle blending without causing text errors. so two needs can be satisfied. i'm guessing the text priority part is already exist and fullfilled its purpose. because fighting games like this could benefit more from graphics than text.

**Comment by hizzlekizzle** [comment]
It's possible, but it's not just text that gets blended with false positives; any small detail like eyes can get averaged away.

**Comment by whatwwkd4496** [comment]
yes please.

**Comment by whatwwkd4496** [comment]
i've been thinking. this dithering is basicly checkerboard rendering of two layer of frames that overlays on top of each other to produce transparency effect, so it's two half-pixel-frames that needs to be upscaled to full frames and overlaid on top of each. so we already have upscaling filters like scalex, xbr, and nnedi3. first we need to upscale odd and even pixels separately then blend them to a single frame. is anyone willing to attempt that approach?

**Comment by whatwwkd4496** [comment]
i found something good
![Eternal Champions (USA)-210627-205759-1x_DitherDeleterV3-Smooth- 32 _115000_G_4x](https://user-images.githubusercontent.com/29114898/123745448-80c81b00-d8da-11eb-9924-7a9ee9d28a52.png)
but its an ESRGAN model (DitherDeleterV3) and rendering time is terrible.
